### PR TITLE
chore(atomic): support i18n json compatibility v4

### PR DESCRIPTION
**Pull request by dmbrooke**
https://coveord.atlassian.net/browse/KIT-2383

Change is a lot smaller than it appears, but a majority of the changes come from the duplicate entries in locales.json to appease the new json plural format.

Changes:
* ~Add [polyfill](https://github.com/eemeli/intl-pluralrules) for [Intl.PluralRules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/PluralRules) to ensure v4 json compatibility works should they chose to upgrade.~ Never mind, Intl.PluralRules is supported across all of the same browsers we support. This polyfill won't be necessary.
* Add atomic-search-interface prop for implementers to specify which version to use (v3 by default)
* Log warning when v3 is detected upon changing languages
* UT to ensure correct translations are used for each version.

### CODE REVIEW

File: packages/atomic/src/components/common/interface/interface-common.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ export interface BaseAtomicInterface<EngineType extends AnyEngineType>
   host: HTMLStencilElement;
   bindings: AnyBindings;
   error?: Error;
+  compatibilityJson: 'v1' | 'v2' | 'v3' | 'v4';
```
Comment (alexprudhomme) [inline packages/atomic/src/components/common/interface/interface-common.tsx:29]: Why is v1 and v2 here ? Should this be `i18nCompatibilityJSONVersion` ?

**Review by alexprudhomme — APPROVED** [review]
LGTM !

**Review by olamothe — CHANGES_REQUESTED** [review]
I wonder if there's something we could do when we detect an entry override/entry getting added to the i18n dictionary that follows the old format (ie: warn) ?

File: packages/atomic/src/components/common/interface/interface-common.tsx (hunk 1 above)
Comment (dmbrooke) [reply packages/atomic/src/components/common/interface/interface-common.tsx:29]: Whoops, could've sworn I removed 'v1' | 'v2' at one point, thanks for pointing this out! 

Also I agree, that name is more clear. I'll use that instead.

**Comment by dmbrooke** [comment]
> I wonder if there's something we could do when we detect an entry override/entry getting added to the i18n dictionary that follows the old format (ie: warn) ?

After doing some research, I'm not convinced that this could be done cleanly without either having to introduce a breaking change to `interface-common` or updating the localization docs...

Ideally, we would create our own class that extends the i18n library class, and then overload the one method we need (`addResourceBundle`) to add a warning before calling the original method. _**However**_, because i18n is actually an interface, I believe we would also need to duplicate all of the other methods that are called within `addResourceBundle`, creating a whole bunch of code duplication.

So for now, I think I'll keep it simple and log a warning whenever v3 is detected on the execution of the first search 🙂

**Review by olamothe — APPROVED** [review]

**Review by louis-bompart — CHANGES_REQUESTED** [review]
Why polyfill?
API seems supported on all our supported browser, so I'd say nay to a polyfill?

**Comment by dmbrooke** [comment]
> Why polyfill? API seems supported on all our supported browser, so I'd say nay to a polyfill?

Oh interesting, do we have an official list of supported browsers somewhere for Atomic?

I added the polyfill as a safety net (which is suggested in the [i18next migration guide](https://www.i18next.com/misc/migration-guide#v20.x.x-to-v21.0.0)) in case there was a context in which `PluralRules` didn't exist. Although now that I think about it, this is Atomic (not Headless), so we have a much clearer idea as to where this code will be executing, so you're probably right about not needing it 🙂

**Review by louis-bompart — APPROVED** [review]

**Review by jelmedini — APPROVED** [review]

**Pull request merged**